intro-js.md

```diff
@@ -38,6 +38,13 @@ Originally they were implemented by creating multiple functions.
 
 Classes have now been provided as part of [ECMAScript 2015](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes), but this is just syntactic sugar. 
 
+#### Side Note
+What exactly is syntactic sugar? 
+
+Syntactic sugar is basically syntax for a programming language that makes it easier for developers to read and write. 
```

Comment:
We need something a little bit more in-depth than this. 
What I was thinking was more along the lines of:
First introduce ECMAScript, the different versions, then explain how compatibility of each version is dependent on the browser. 
After that, we can mention that syntactic sugar is `helpful syntax to make developer's lives easier for development, but in the end, in terms of functionality differences, there is none`